<launch>
  <arg name="edumip_name"/>
  <arg name="edumip_num"/>

  <node pkg="robot_state_publisher" type="state_publisher" name="robot_state_publisher" output="screen"/>

  <!-- Launch edumip_nav_publisher custom node -->
  <node pkg="edumip_my_robot" type="edumip_nav_publisher" name="edumip_nav_publisher" args="$(arg edumip_name)/edumip_body"/>


  <!-- Launch Stabilizer Nodes -->
  <node pkg="edumip_my_robot" type="edumip_tf_stabilizer" name="edumip_tf_stabilizer" args="$(arg edumip_name)/edumip_body $(arg edumip_name)/edumip_laser"/>

  <!-- Launch Laser Scan Node -->
  <node pkg="tf_laser" type="tf_laser" name="tf_laser" args="$(arg edumip_num) edumip_scan"/>

  <!-- Launch move base -->
    <node pkg ="move_base" type="move_base" respawn="false" name="move_base" output="screen">
      <rosparam file="$(find edumip_my_robot)/costmap/$(arg edumip_name)/costmap_common_params.yaml" command="load" ns="global_costmap"/>
      <rosparam file="$(find edumip_my_robot)/costmap/$(arg edumip_name)/costmap_common_params.yaml" command="load" ns="local_costmap"/>
      <rosparam file="$(find edumip_my_robot)/costmap/$(arg edumip_name)/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find edumip_my_robot)/costmap/$(arg edumip_name)/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find edumip_my_robot)/costmap/base_local_planner_params.yaml" command="load" />
    </node>

</launch>